tests/test_apply.py

```diff
@@ -569,6 +569,14 @@ def test_load_default_qubit_device(self):
         assert dev.shots is None
         assert dev.short_name == "lightning.gpu"
 
+    def test_with_shots(self):
+        """Test that lightning.gpu does not support finite shots"""
+
+        with pytest.raises(
+            ValueError, match="lightning.gpu does not support finite shots, please use shots=None"
+        ):
+            qml.device("lightning.gpu", wires=2, shots=1)
+
```

Comment:
Adding a test for `shots=0` and `shots=None` too and checking if no exceptions raised should be good.